packages/in_app_purchase/ios/Classes/FIAPaymentQueueHandler.m

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FIAPaymentQueueHandler.h"
+
+NSString *const TestingProductID = @"testing";
+
+@interface FIAPaymentQueueHandler ()
+
+@property(strong, nonatomic) SKPaymentQueue *queue;
+@property(nullable, copy, nonatomic) TransactionsUpdated transactionsUpdated;
+@property(nullable, copy, nonatomic) TransactionsRemoved transactionsRemoved;
+@property(nullable, copy, nonatomic) RestoreTransactionFailed restoreTransactionFailed;
+@property(nullable, copy, nonatomic)
+    RestoreCompletedTransactionsFinished paymentQueueRestoreCompletedTransactionsFinished;
+@property(nullable, copy, nonatomic) ShouldAddStorePayment shouldAddStorePayment;
+@property(nullable, copy, nonatomic) UpdatedDownloads updatedDownloads;
+
+@property(strong, nonatomic) NSMutableDictionary *transactionsSetter;
+
+@end
+
+@implementation FIAPaymentQueueHandler
+
+- (instancetype)initWithQueue:(nonnull SKPaymentQueue *)queue
+                     transactionsUpdated:(nullable TransactionsUpdated)transactionsUpdated
+                      transactionRemoved:(nullable TransactionsRemoved)transactionsRemoved
+                restoreTransactionFailed:(nullable RestoreTransactionFailed)restoreTransactionFailed
+    restoreCompletedTransactionsFinished:
+        (nullable RestoreCompletedTransactionsFinished)restoreCompletedTransactionsFinished
+                   shouldAddStorePayment:(nullable ShouldAddStorePayment)shouldAddStorePayment
+                        updatedDownloads:(nullable UpdatedDownloads)updatedDownloads {
+  self = [super init];
+  if (self) {
+    self.queue = queue;
+    self.transactionsUpdated = transactionsUpdated;
+    self.transactionsRemoved = transactionsRemoved;
+    self.restoreTransactionFailed = restoreTransactionFailed;
+    self.paymentQueueRestoreCompletedTransactionsFinished = restoreCompletedTransactionsFinished;
+    self.shouldAddStorePayment = shouldAddStorePayment;
+    self.updatedDownloads = updatedDownloads;
+    self.transactionsSetter = [NSMutableDictionary new];
+  }
+  return self;
+}
+
+- (void)addPayment:(SKPayment *)payment {
+  NSString *productID = payment.productIdentifier;
+  if (self.testing) {
+    productID = TestingProductID;
+  }
+  [self.queue addPayment:payment];
+}
+
+- (void)finishTransaction:(SKPaymentTransaction *)transaction {
+  @try {
+    [self.queue finishTransaction:transaction];
+  } @catch (NSException *e) {
```

Comment:
What's the benefit to catching and then immediately `raise`ing the exception again here?

Reply:
There is no benefit here, I should remove this.